src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -4659,104 +4529,210 @@ namespace EvaporativeCoolers {
-                        if (ZoneEvapUnit(UnitNum).FanType_Num != DataHVACGlobals::FanType_SystemModelObject) {
-                            Fans::SimulateFanComponents(
-                                ZoneEvapUnit(UnitNum).FanName, false, ZoneEvapUnit(UnitNum).FanIndex, _, ZoneCompTurnFansOn, ZoneCompTurnFansOff);
-                        } else {
-                            HVACFan::fanObjs[ZoneEvapUnit(UnitNum).FanIndex]->simulate(_, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
-                        }
-                    }
+        // REFERENCES:
+        // na
 
-                    if (ZoneEvapUnit(UnitNum).EvapCooler_1_AvailStatus) {
-                        SimEvapCooler(ZoneEvapUnit(UnitNum).EvapCooler_1_Name, ZoneEvapUnit(UnitNum).EvapCooler_1_Index);
-                    }
+        // Using/Aliasing
+        using DataHVACGlobals::ZoneCompTurnFansOff;
+        using DataHVACGlobals::ZoneCompTurnFansOn;
 
-                    if ((ZoneEvapUnit(UnitNum).EvapCooler_2_Index > 0) && ZoneEvapUnit(UnitNum).EvapCooler_2_AvailStatus) {
-                        SimEvapCooler(ZoneEvapUnit(UnitNum).EvapCooler_2_Name, ZoneEvapUnit(UnitNum).EvapCooler_2_Index);
-                    }
-                    if (ZoneEvapUnit(UnitNum).FanLocation == DrawThruFan) {
-                        if (ZoneEvapUnit(UnitNum).FanType_Num != DataHVACGlobals::FanType_SystemModelObject) {
-                            Fans::SimulateFanComponents(
-                                ZoneEvapUnit(UnitNum).FanName, false, ZoneEvapUnit(UnitNum).FanIndex, _, ZoneCompTurnFansOn, ZoneCompTurnFansOff);
-                        } else {
-                            HVACFan::fanObjs[ZoneEvapUnit(UnitNum).FanIndex]->simulate(_, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
-                        }
-                    }
-                }
+        // Locals
+        // SUBROUTINE ARGUMENT DEFINITIONS:
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        // na
+        // INTERFACE BLOCK SPECIFICATIONS:
+        // na
+
+        // DERIVED TYPE DEFINITIONS:
+        // na
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Real64 MinHumRat; // minimum humidity ratio
+
+        int const ZoneNodeNum = ZoneEvapUnit(UnitNum).ZoneNodeNum;
+        int const OAInletNodeNum = ZoneEvapUnit(UnitNum).OAInletNodeNum;
+        int const OutletNodeNum = ZoneEvapUnit(UnitNum).UnitOutletNodeNum;
+        int const ReliefNodeNum = ZoneEvapUnit(UnitNum).UnitReliefNodeNum;
+        int const FanInletNodeNum = ZoneEvapUnit(UnitNum).FanInletNodeNum;
+        int const FanOutletNodeNum = ZoneEvapUnit(UnitNum).FanOutletNodeNum;
+        int const EvapCooler_1_Index = ZoneEvapUnit(UnitNum).EvapCooler_1_Index;
+        int const EvapCooler_2_Index = ZoneEvapUnit(UnitNum).EvapCooler_2_Index;
+
+        // calculate unit sensible cooling output
+        if (PartLoadRatio > 0) {
+            Node(OAInletNodeNum).MassFlowRate = ZoneEvapUnit(UnitNum).DesignAirMassFlowRate * PartLoadRatio;
```

Comment:
PLR is used to modulate the supply air flow. Prior to this fix, the avp units always run at full flow, and some times over-cooled the space. This is one of the main request of the user, avoiding over cool.